### DISCUSSION TITLE
Another language

### DISCUSSION THREAD

**Discussion by ChaoRuk** [open-mmlab/mmocr#1488] [General]
if i would like to train model that no english (another language). what do i have to do ?

**Comment by gaotongxiao** [comment]
The answer depends on MMOCR's version. If you are new to MMOCR, I strongly recommend you start with [MMOCR 1.0](https://github.com/open-mmlab/mmocr/tree/1.x) since it has significant changes, improvements and more detailed [documentation](https://mmocr.readthedocs.io/en/dev-1.x/), compared to 0.x.

## MMOCR 0.x

In MMOCR 0.x, you need to prepare a dictionary file where characters are listed line by line ([example dict](https://download.openmmlab.com/mmocr/textrecog/sar/dict_printed_chinese_english_digits.txt)). Then update `lable_convertor`'s config as something similar to https://github.com/open-mmlab/mmocr/blob/5fc920495acf3e1e1c933f85b284f7384481bf71/configs/textrecog/sar/sar_r31_parallel_decoder_chinese.py#L6-L11

Sometimes you may also need to revise the model architecture to ensure the output channel of the final projection layer matches the dictionary size. This variable is usually named [`num_chars`](https://github.com/open-mmlab/mmocr/blob/5fc920495acf3e1e1c933f85b284f7384481bf71/configs/_base_/recog_models/abinet.py#L1-L5). 

## MMOCR 1.0

Prepare a dictionary file and replace the `Dictionary`'s `dict_file` field with its path ([example](https://github.com/gaotongxiao/mmocr/blob/f30c16ce96bd2393570c04eeb9cf48a7916315cc/configs/textrecog/abinet/_base_abinet-vision.py#L3)). The model architecture will be automatically updated without any manual effort.
